opm/output/data/Aquifer.hpp

```diff
@@ -96,6 +178,36 @@ namespace Opm { namespace data {
         return 0.0;
     }
 
+    inline bool AquiferData::operator==(const AquiferData& other) const
+    {
+        const auto equal_structure =
+            (this->aquiferID == other.aquiferID) &&
+            (this->pressure == other.pressure) &&
+            (this->fluxRate == other.fluxRate) &&
+            (this->volume == other.volume) &&
+            (this->initPressure == other.initPressure) &&
+            (this->datumDepth == other.datumDepth) &&
+            (this->type == other.type) &&
+            ((this->aquFet == nullptr) == (other.aquFet == nullptr)) &&
+            ((this->aquCT == nullptr) == (other.aquCT == nullptr)) &&
+            ((this->aquFet == nullptr) != (this->aquCT == nullptr));
```

Comment:
This last check looks more like an internal structure check than a comparison?

Reply:
> This last check looks more like an internal structure check than a comparison

It sort of is, although this coupled with the two checks before ensure that both `this` and `other` have that consistent structure (exactly one of `aquFet` and `aquCT` is non-null).  We also assign the result to an object named [`equal_structure`](https://github.com/OPM/opm-common/blob/769e69bedc5149872d8ea0353161b4734588e74c/opm/output/data/Aquifer.hpp#L183) which is supposed to imply structural integrity too.